docs/develop/dotnet/data-encryption.mdx

```diff
@@ -26,26 +26,109 @@ This page shows the following:
+}
+```
+
+**Set Data Converter to use custom Payload Codec**
+
+When creating a client, the default `DataConverter` can be updated with the payload codec like so:
+
+```csharp
+var myClient = await TemporalClient.ConnectAsync(new("localhost:7233")
+{
+    DataConverter = DataConverter.Default with { PayloadCodec = new EncryptionCodec() },
+});
+```
 
 ## Payload conversion
 
-<!-- CONTENT-->
+Temporal SDKs provide a default [Payload Converter](/dataconversion#payload-converter) that can be customized to convert a custom data type to [Payload](/dataconversion#payload) and back.
 
 ### Conversion sequence {#conversion-sequence}
 
-<!-- CONTENT-->
+The order in which your encoding Payload Converters are applied depend on the order given to the Data Converter.
+You can set multiple encoding Payload Converters to run your conversions.
+When the Data Converter receives a value for conversion, it passes through each Payload Converter in sequence until the converter that handles the data type does the conversion.
 
 ### Custom Payload Converter {#custom-payload-converter}
 
 **How to use a custom Payload Converter with the .NET SDK.**
 
-<!-- CONTENT-->
-
-**List of default Data Converter supports converting multiple types**
-
-<!-- CONTENT-->
+Data converters are used to convert raw Temporal payloads to/from actual .NET types.
+A custom data converter can be set via the `DataConverter` option when creating a client. Data converters are a combination of payload converters, payload codecs, and failure converters.
+Payload converters convert .NET values to/from serialized bytes. Payload codecs convert bytes to bytes (e.g. for compression or encryption). Failure converters convert exceptions to/from serialized failures.
+
+Data converters are in the `Temporalio.Converters` namespace.
+The default data converter uses a default payload converter, which supports the following types:
+
+- `null`
+- `byte[]`
+- `Google.Protobuf.IMessage` instances
+- Anything that `System.Text.Json` supports
+- `IRawValue` as unconverted raw payloads
+
+Custom converters can be created for all uses. For example, to create client with a data converter that converts all C#
+property names to camel case, you would:
+
+```csharp
+using System.Text.Json;
+using Temporalio.Client;
+using Temporalio.Converters;
+
+public class CamelCasePayloadConverter : DefaultPayloadConverter
+{
+    public CamelCasePayloadConverter()
+      : base(new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase })
+    {
+    }
+}
+
+var client = await TemporalClient.ConnectAsync(new()
+{
+    TargetHost = "localhost:7233",
+    Namespace = "my-namespace",
+    DataConverter = DataConverter.Default with { PayloadConverter = new CamelCasePayloadConverter() },
+});
+```
 
 ### Custom Type Data Conversion
 
-<!-- CONTENT-->
+<!--  TODO(cretz): Develop https://github.com/temporalio/samples-dotnet/issues/38 first -->
```

Comment:
We can remove this section pending that sample, or if it's urgent I can prioritize it